### PR TITLE
Add hint for unexpected type definition (td)

### LINKED ISSUE #2267: td, to: Does not print any errors
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Windows x64 64
| File format of the file you reverse (mandatory)      | PE
| Architecture/bits of the file (mandatory)            | x64
| `rizin -v` full output, **not truncated** (mandatory)  | rizin 0.4.0-git @ windows-x86-64 commit: f2e421a896e8502724b36d427b9913ca900516c6, build: 29/01/2022__11:48:41.71

### Expected behavior
On failing to parse, or any other error, the input of `td`, `to`, I expect to get an error saying what's wrong.

### Actual behavior
It fails silently, making it hard to understand what's wrong or even notice that anything is wrong in the first place. There isn't even any debug message with `e log.level=0`.

### Steps to reproduce the behavior
1. `td "stract Foo { int Bar; }"`
2. ` ts~Foo`
3. `td "struct Foo { int Bar; }"`
4. `ts~Foo`

### Additional Logs, screenshots, source code,  configuration dump, ...

screenshot

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Add hint for unexpected type definition. 

...

**Test plan**

...

**Closing issues**

closes https://github.com/rizinorg/rizin/issues/2267
...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: librz/type/parser/c_cpp_parser.c [hunk 1]
```diff
@@ -168,10 +180,12 @@ static int type_parse_string(CParserState *state, const char *code, char **error
 	for (i = 0; i < root_node_child_count; i++) {
 		TSNode child = ts_node_named_child(root_node, i);
 		// We skip ";" or "," - empty expressions
-		const char *node_type = ts_node_type(child);
-		if (!strcmp(node_type, "expression_statement")) {
```
Comment (ret2libc) [inline librz/type/parser/c_cpp_parser.c:172]: don't you need to check this anyway?

Comment (PeiweiHu) [reply librz/type/parser/c_cpp_parser.c:172]: It seems they are used to skip `, `and `;`, maybe using the result of `ts_node_sub_string()` is better. The original way will also skip the statement like `td "stract;"`.

Comment (ret2libc) [reply librz/type/parser/c_cpp_parser.c:172]: @XVilka ?

Comment (XVilka) [reply librz/type/parser/c_cpp_parser.c:172]: Yes, it was used for any empty expression: https://github.com/rizinorg/tree-sitter-c/blob/bare-types-support/grammar.js#L606

I think this change is OK.

**Pull request merged**